src/com/gfredericks/schpec/number.clj

```diff
@@ -0,0 +1,119 @@
+(ns com.gfredericks.schpec.number
+  (:require [clojure.spec :as s]
+            [clojure.spec.gen :as gen])
+  (:import [java.math BigDecimal MathContext RoundingMode]))
+
+(defn real?
+  [x]
+  (and (number? x)
+       (case x
+         Double/POSITIVE_INFINITY false
+         Double/NEGATIVE_INFINITY false
+         Double/NaN false
+         true)))
+
+(s/def ::real
+  (s/spec real?
+          :gen #(gen/double* {:infinite? false :NaN? false})))
```

Comment:
yes, irrational numbers are the sort of thing I'm thinking of w.r.t. the word "real"